File: .agent/scripts/strip_tribunal.py
import os
import re

def strip_boilerplate(file_path):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    original_len = len(content)

    # Regex to find everything from "## 🏛️ Tribunal Integration" (or without emojis) to the end of the file or next major H1
    # Often it appears at the end of agents
    content = re.sub(r'## 🏛️ Tribunal Integration[\s\S]*?(?=\n# |\Z)', '', content)
    content = re.sub(r'## Tribunal Integration[\s\S]*?(?=\n# |\Z)', '', content)
    
    # Strip Cross-Workflow Navigation since GEMINI.md has global handling
    content = re.sub(r'## Cross-Workflow Navigation[\s\S]*?(?=\n# |\Z)', '', content)

    # Strip What the Maker Is Not Allowed to Do (it's in GEMINI.md)
    content = re.sub(r'## What the Maker Is Not Allowed to Do[\s\S]*?(?=\n# |\Z)', '', content)

    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(content.strip() + '\n')

    return original_len, len(content)

def main():
    dirs_to_check = ['.agent/agents', '.agent/workflows']
    total_stripped = 0

    for d in dirs_to_check:
        if not os.path.exists(d): continue
        for file in os.listdir(d):
            if file.endswith('.md'):
                file_path = os.path.join(d, file)
                orig, new = strip_boilerplate(file_path)
                total_stripped += (orig - new)

    print(f"Stripped {total_stripped} bytes of repetitive boilerplate.")

if __name__ == '__main__':
    main()

File: .agent/scripts/strip_tribunal.js
#!/usr/bin/env node
/**
 * strip_tribunal.js
 */
'use strict';

const fs = require('fs');
const path = require('path');

function stripBoilerplate(filePath) {
    const original = fs.readFileSync(filePath, 'utf8');
    const originalLen = Buffer.byteLength(original, 'utf8');

    let content = original;
    content = content.replace(/## 🏛️ Tribunal Integration[\s\S]*?(?=\n# |\Z)/g, '');
    content = content.replace(/## Tribunal Integration[\s\S]*?(?=\n# |\Z)/g, '');
    content = content.replace(/## Cross-Workflow Navigation[\s\S]*?(?=\n# |\Z)/g, '');
    content = content.replace(/## What the Maker Is Not Allowed to Do[\s\S]*?(?=\n# |\Z)/g, '');

    fs.writeFileSync(filePath, content.trim() + '\n', 'utf8');
    const newLen = Buffer.byteLength(content.trim() + '\n', 'utf8');

    return [originalLen, newLen];
}

function main() {
    const dirsToCheck = ['.agent/agents', '.agent/workflows'];
    let totalStripped = 0;

    for (const d of dirsToCheck) {
        if (!fs.existsSync(d)) continue;
        const files = fs.readdirSync(d);
        for (const file of files) {
            if (file.endsWith('.md')) {
                const filePath = path.join(d, file);
                const [orig, newL] = stripBoilerplate(filePath);
                totalStripped += (orig - newL);
            }
        }
    }

    console.log(`Stripped ${totalStripped} bytes of repetitive boilerplate.`);
}

if (require.main === module) {
    main();
}
